### PR TITLE
initial commit for meta git remote

### PR DESCRIPTION
**Pull request by mateodelnorte**
- [x] git remote -v
- [x] git remote show origin
- [x] git remote rename oldname newname
- [x] git remote rm name

### CODE REVIEW

File: bin/meta-git-remote [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+if (process.argv[2] === '--help')
+  return console.log(usages);
+
+let first = process.argv[2] === 'blank' ?
+                      process.argv[3] || '':
+                      process.argv[2] || '';
+
+const subCommandParsers = {
+  '-v': function (argv) { return '' },
```
Comment (mikelax) [inline bin/meta-git-remote]: hey @mateodelnorte shouldn't this line be return '-v'
`git remote` and `git remote -v` do return different output.

File: bin/meta-git-remote [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  },
+  'rename': function (argv) {
+    
+  },
+  'rm': function (argv) {}
+}
+
+const subCommandParser = subCommandParsers[first];
+
+if ( ! subCommandParser) return console.log(usages);
```
Comment (mikelax) [inline bin/meta-git-remote]: what if the user just runs `meta git remote` won't this line cause the usages to be displayed as there is no sub-command? Is that what you want or do you want to allow the user to run just `git remote`?

File: bin/meta-git-remote (hunk 1 above)
Comment (mateodelnorte) [reply bin/meta-git-remote]: yep. that was a typo

File: bin/meta-git-remote (hunk 2 above)
Comment (mateodelnorte) [reply bin/meta-git-remote]: Probably make sense to force them to type `meta git remote --help` for usage, in this case. I really just wanted to get a skeleton out there to show you how easy it is to make these commands and see if you'd be interested in helping to develop them. I put this together in just a couple minutes. I'd bet you'd be able to get an api you'd like equally quickly. Incremental push coming momentarily. 

Happy to make you a contributor, if you're interested, and we can just go back and forth with commits on the PR until we think it looks good.

Comment (mikelax) [reply bin/meta-git-remote]: thanks. I should be able to find some time to see if I can work on this ticket and then look into the gh plugin. 
One question on this command. How come we have to do the parsing and logic for detection of each of the sub commands? can't we let the actual `git remote` command handle that and if there is an error in user input the default help will be displayed? 

```
Mike@Michaels-MacBook-Pro:chef (master) $ git remote foo
error: Unknown subcommand: foo
usage: git remote [-v | --verbose]
   or: git remote add [-t <branch>] [-m <master>] [-f] [--tags | --no-tags] [--mirror=<fetch|push>] <name> <url>
   or: git remote rename <old> <new>
   or: git remote remove <name>
   or: git remote set-head <name> (-a | --auto | -d | --delete | <branch>)
   or: git remote [-v | --verbose] show [-n] <name>
   or: git remote prune [-n | --dry-run] <name>
   or: git remote [-v | --verbose] update [-p | --prune] [(<group> | <remote>)...]
   or: git remote set-branches [--add] <name> <branch>...
   or: git remote get-url [--push] [--all] <name>
   or: git remote set-url [--push] <name> <newurl> [<oldurl>]
   or: git remote set-url --add <name> <newurl>
   or: git remote set-url --delete <name> <url>

    -v, --verbose         be verbose; must be placed before a subcommand
```

Comment (mateodelnorte) [reply bin/meta-git-remote]: Yea, you could just as easily do that. That said, you do need to detect --help and give *some* sort of help text. If you pass everything to the command, you'll end up showing `git remote --help` for every child repo.

**Comment by mateodelnorte** [comment]
Just pushed a change to pass full sub command text to the child command, as you mentioned. Doing so revealed a good technique for going forward, which will be something along the lines of creating the following: 
```
assertUsage('git remote', `
usage: meta git remote [-v | --verbose]
   or: meta git remote add [-t <branch>] [-m <master>] [-f] [--tags | --no-tags] [--mirror=<fetch|push>] <name> <url>
   or: meta git remote rename <old> <new>
   or: meta git remote remove <name>
   or: meta git remote set-head <name> (-a | --auto | -d | --delete | <branch>)
   or: meta git remote [-v | --verbose] show [-n] <name>
   or: meta git remote prune [-n | --dry-run] <name>
   or: meta git remote [-v | --verbose] update [-p | --prune] [(<group> | <remote>)...]
   or: meta git remote set-branches [--add] <name> <branch>...
   or: meta git remote get-url [--push] [--all] <name>
   or: meta git remote set-url [--push] <name> <newurl> [<oldurl>]
   or: meta git remote set-url --add <name> <newurl>
   or: meta git remote set-url --delete <name> <url>

    -v, --verbose         be verbose; must be placed before a subcommand`);
```

Where assert usage can parse the given usage text, compare the given process.argv values against the possible usages, send back `--help` when there is a mismatch, or allow execution when passed. 

For now, I'm just doing a poor man's version of spot checking some usages, but it would be great to have something like the above where you can easily create plugins by getting the usage from the original command, drop it in, and execute using loop when things pass.

File: bin/meta-git-remote [hunk 3]
```diff
@@ -7,40 +7,45 @@ const loop = require('loop-things');
...
+   or: meta git remote get-url [--push] [--all] <name>
+   or: meta git remote set-url [--push] <name> <newurl> [<oldurl>]
+   or: meta git remote set-url --add <name> <newurl>
+   or: meta git remote set-url --delete <name> <url>
+
+    -v, --verbose         be verbose; must be placed before a subcommand`;
+
+if (process.argv[2] === '--help') return console.log(usages);
+
+var valid = new Set();
```
Comment (mikelax) [inline bin/meta-git-remote]: You can change this to a `const` but otherwise :shipit:

**Comment by mateodelnorte** [comment]
Created a first draft of a function to check input against known usage. Thoughts?

**Comment by mikelax** [comment]
hey @mateodelnorte sorry I just got a chance to take a look at this. 
The function check looks good, I had a few stylistic comments, variable names, etc but function looks good. :shipit:

**Comment by mateodelnorte** [comment]
Word. Shipped, it is. ;) Thanks

**Pull request merged**